fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java

```diff
@@ -1901,8 +1905,9 @@ public int getAsInt() {
                 AddFieldClause addFieldClause = (AddFieldClause) alterClause;
                 modifyFieldColumns = Set.of(addFieldClause.getColName());
                 checkModifiedColumWithMaterializedViews(olapTable, modifyFieldColumns);
-  
-                int id = colUniqueIdSupplier.getAsInt();
+
+                // increase maxColUniqueId and assign the uniqueId to new added field
+                int id = olapTable.incAndGetMaxColUniqueId();
```

Comment:
1. why not update max column unique id using field id when `rebuildFullSchema` as other alters. 
2. if `incAndGetMaxColUniqueId` is used in all alters, including add column or field, then there is no need to update max unique id in `rebuildFullSchema`?
3. `OlapTable.setMaxColUniqueId` needs to ensure that the id does not fall back?

Reply:
> 1. why not update max column unique id using field id when `rebuildFullSchema` as other alters.
> 2. if `incAndGetMaxColUniqueId` is used in all alters, including add column or field, then there is no need to update max unique id in `rebuildFullSchema`?

1. Because we only check the uniqueid of column but not check the fileld of struct field in `rebuildFullSchema`.